src/settings/LocationLocations/LocationManager.js

```diff
@@ -374,47 +485,203 @@ class LocationManager extends React.Component {
-        servicePointsByName={this.state.servicePointsByName}
-        servicePointsById={this.state.servicePointsById}
-        parseInitialValues={this.parseInitialValues}
-        entryLabel={this.props.intl.formatMessage({ id: 'ui-tenant-settings.settings.location.locations.location' })}
-        entryFormComponent={LocationForm}
-        validate={this.validate}
-        asyncValidate={this.asyncValidate}
-        asyncBlurFields={['name', 'code']}
-        rowFilter={this.renderFilter()}
-        rowFilterFunction={this.filterRow}
-        nameKey="name"
-        permissions={{
-          put: 'settings.tenant-settings.enabled',
-          post: 'settings.tenant-settings.enabled',
-          delete: 'settings.tenant-settings.enabled',
+  onSave = location => {
+    const { match } = this.props;
+
+    const action = location.id ? 'PUT' : 'POST';
+
+    return this.props.mutator.entries[action](location)
+      .then(updatedLocation => {
+        this.transitionToParams({
+          _path: `${match.path}/${updatedLocation.id}`,
+          layer: null,
+        });
+        this.setState({ selectedId: updatedLocation.id });
+      })
+      .catch(error => this.showSubmitErrorCallout(error.message || error.statusText));
+  };
+
+  showCalloutMessage(name) {
+    if (!this.callout.current) return;
+
+    const message = (
+      <SafeHTMLMessage
+        id="stripes-core.successfullyDeleted"
+        values={{
+          entry: this.entryLabel,
+          name: name || '',
         }}
       />
     );
+
+    this.callout.current.sendCallout({ message });
+  }
+
+  showSubmitErrorCallout(error) {
+    if (!this.callout.current) return;
+
+    this.callout.current.sendCallout({
+      type: 'error',
+      message: error,
+    });
+  }
+
+  render() {
+    const {
+      match,
+      label,
+      location,
+    } = this.props;
+    const {
+      institutionId,
+      campusId,
+      libraryId,
+      sort,
+      sortDir,
+      selectedId,
+      servicePointsById,
+      servicePointsByName,
+    } = this.state;
+
+    const locations = this.prepareLocationsData();
+    const contentData = locations.filter(row => row.libraryId === libraryId);
+    const query = location.search ? queryString.parse(location.search) : {};
+    const defaultEntry = { isActive: true, institutionId, campusId, libraryId, servicePointIds: [{ selectSP: '', primary: true }] };
+    const adding = location.search.match('layer=add');
+
+    const selectedItem = (selectedId && !adding)
+      ? find(contentData, entry => entry.id === selectedId) : defaultEntry;
+
+    const initialValues = this.parseInitialValues(selectedItem);
+
+    const container = document.getElementById('ModuleContainer');
+
+    if (!container) return (<div />);
+
+    return (
+      <Paneset
+        defaultWidth="fill"
+        data-test-entry-manager
+      >
+        <Pane
+          defaultWidth="fill"
+          paneTitle={label}
+        >
+          <SearchAndSortQuery>
```

Comment:
@VictorSoroka1 very nice to see `<SearchAndSortQuery>` being used here!

Reply:
Thank you @mkuklis, it really fit there)